src/item.cpp

```diff
@@ -5838,13 +5838,12 @@ std::string item::get_plant_name() const
 
 bool item::is_dangerous() const
 {
-    // Note: doesn't handle the pipebomb or radio bombs
-    // Consider flagging dangerous items with an explicit flag instead
-    static const std::string explosion_string( "explosion" );
-    if( type->can_use( explosion_string ) ) {
+    if( has_flag( "DANGEROUS" ) ) {
         return true;
     }
 
+    // Note: Item should be dangerous regardless of what type of a container is it
+    // Visitable interface would skip some options
```

Comment:
> In what circumstances would that be a problem?

Any case where contents are processed and could be dangerous.
Currently no such option is possible, but realistically:
* Grenade launcher magazine with a dud/activated time bomb
* Radio mine launcher with radio mine loaded into it. Or a magazine for it.
* Overheating battery
* Grenade stored in a sealed can